### ISSUE TITLE
Blood and morphine does not work correctly [Medical Basic]

### ISSUE THREAD

**Issue by RaiderSpectre** [acemod/ACE3#4421]
**Arma 3 Version:** 1.62.137494 (stable)
**CBA Version:** 3.0.0.160713 (stable)
**ACE3 Version:** 3.7.0 (stable)

**Mods:**
- `@CBA_A3`
- `@ace`

**Description:**
the value of blood and pain does not change when injecting moprhine and blood in a patient, as result if the patient is greatly injured it will never be healed.
If the patient injects himself with morhine pain will decrease.

**Steps to reproduce:**
https://www.youtube.com/watch?v=tbJX0F1VpcE

**Where did the issue occur?**
- server dedicated multiplayer

**Placed Modules:**
https://cloud.githubusercontent.com/assets/22255802/18608033/2587ffe2-7cde-11e6-82af-862c98eb8f43.jpg

**RPT log file:**
https://github.com/acemod/ACE3/files/478143/arma3server1_2016-09-17_14-20-44.rpt.txt

**Comment by alganthe** [comment]
Can you confirm the issue here:
https://github.com/alganthe/ganthe_ACE3medical_vr

- Do a multiplayer preview.
- Enable debug in the mission params.
- Spawn a unit via the terminal and enable debug on it via a self interaction.

**Comment by alganthe** [comment]
I can NOT confirm the issue, in basic blood and pain are incremented and decremented properly.

**Comment by RaiderSpectre** [comment]
Hi alganthe, i tested your mission but is for Advanced medical System,  the issue i reporter is for the basic medical system. i have no idea if there is a correlation between blood and pain for basic and advanced but i can confirm the problem still persist.

**Comment by alganthe** [comment]
Make sure to switch the modules to basic, that's what I've done.
It might be only happening in MP, test it with your buddy on the mission (place a second unit and switch the modules) and enable debug

**Comment by nomisum** [comment]
Didnt test properly, but we had similar issues in our mission last night. 2 guys died in agony and pain despite having received tons of blood and morphine.

**Comment by RaiderSpectre** [comment]
after more test, it appears the patient needs 6/7 liters of blood to restore the "has lost a lot of blood" condition, but the morphine seems to work randomly in some case the pain is restored in other persist.

**Comment by gerseun** [comment]
We have the same issue also with the new version. We noticed that reducing the blood loss coefficient resolve the issue partially but when awake the stamina drop down to zero after few steps.

**Comment by commy2** [comment]
Yes. The blood volume / loss is not using realistic values at all. It's just a value between 0 and 100 and you get knocked out at 30 and die at 20.

I changed this in the medical-rewrite to actually use 6 liters as body maximum and to use Class I to IV "Hemorrhage" from https://en.wikipedia.org/wiki/Advanced_trauma_life_support for the effects.

Close since this is/will be fixed in medical-rewrite?

**Comment by commy2** [comment]
Btw. 3.7.0 doesn't change anything about the medical system, so no wonder nothing changed there.

**Comment by flexgrip** [comment]
Is there a temporary fix for this? Or is the medical rewrite able to be compiled yet? This is really putting a damper on our ops because we don't use respawns.

**Comment by RaiderSpectre** [comment]
any chance to resolve this bug? it still persist (it's 5 months now)

**Comment by flexgrip** [comment]
@RaiderSpectre - I was just thinking about this one earlier. I wonder if we can workaround it by editing the bleeding threshold. I can't remember, but it's in the ace medical module at the bottom. It's defaulted to 1 and says that its the threshold to modify bleeding. Maybe we can crank this up or down and fix it?

Unfortunately I don't have time to test it right now. I wouldn't mind fixing this bug if I had time to dig through the code. But I've never worked in ace before so it'll take some time.

**Comment by GodofMonkeys** [comment]
Same thing here.
Is it can avoid this problem by set blood threshold less than 0.5 or even more?

Another question is this blood threshold is mean your bleed rate, but when you hit by some .50 or RPG, you probably will show "has lost a lot of blood" immediately, no matter you set even blood threshold to 0.1 it will same have "has lost a lot of blood".

Is this correct?

**Comment by ViperMaul** [comment]
In the rewrite, this needs testing using the default Medical mode both modes.

**Comment by Whipgit** [comment]
Has there been any news on this? I'm experiencing something similar in multiplayer where the medic is never able to fix the pain or blood loss.

**Comment by tiapfy** [comment]
I'm having the same problem. Is this going to get fixed sooner or later?

**Comment by flexgrip** [comment]
I have a theory on this. I need to double-check it though. Basically, I made a test mission. Spawned an AI, shot him like crazy with a machine gun.

The revive time was set to 600 (ten minutes). So I let the AI lay there and bleed for like 7 minutes. Still 3 minutes left to revive him. And boom. 12+ blood bags. Never was able to revive him.

Now I'm not saying that's realistic or that 100 rounds from a machine gun wouldn't have killed him. But my theory is that although he's being reported as alive, his blood loss and injury had been left so long that he was "*dead*". But the game is still reporting him as alive. So he's somewhere between "prevent instant death" and the revive timer.

So in short, I'm wondering if the code has some issue where a unit can loose all the blood and not ever be revived again, despite being within the revive time limit. So to medics, they see it as a blood bag bug. When it's really that the guy is in a state where he can't be revived.

Again, I will test this again. But if this is true, then we need to look at the code and find out if this is possible.

**Comment by PabstMirror** [comment]
will be fixed in rewrite
